src/core.jl

```diff
@@ -52,8 +52,13 @@ struct JSONRPCEndpoint
 
     err_handler::Union{Nothing,Function}
 
-    function JSONRPCEndpoint(pipe_in, pipe_out, err_handler = nothing)
-        return new(pipe_in, pipe_out, Channel{Any}(Inf), Channel{Any}(Inf), Dict{String,Channel{Any}}(), err_handler)
+    status::Symbol
+
+    read_task::Union{Nothing,Task}
+    write_task::Union{Nothing,Task}
+
+    function JSONRPCEndpoint(pipe_in, pipe_out, err_handler=nothing)
+        return new(pipe_in, pipe_out, Channel{Any}(Inf), Channel{Any}(Inf), Dict{String,Channel{Any}}(), err_handler, :idle, nothing, nothing)
     end
```

Comment:
In general not abusing inner constructor is preferable, right ? As far as I know we only want inner constructor when we want to do some validation at construction.

Reply:
Hm, I have no idea :)